### PR TITLE
[IMP] project_timesheet_time_control: ❤️ 4 UX

### PR DESCRIPTION
**Pull request by yajo**
This is a huge improvement in this module's usability, although basic functionality stays the same. Let me summmarize:

- In `account.analytic.line` views, the *Stop* button is replaced by colorful *Resume/Stop* buttons.
- The new *Resume* button will display a wizard that will duplicate all meaningful fields from the previous `account.analytic.line` record and auto-stop the previous running timer, if any.
- If the previous running timer is very old, some warnings will be displayed.
- In `project.project` and `project.task` views, I added the same buttons (well... it's *Start* instead of *Resume*, because you could not have worked before in that project or task, but it behaves the same).
- If using it from a `project.task`, *Start* will duplicate meaningful fields from your latest line in that task.
- If using it from a `project.project`, *Start* will duplicate meaningful fields your latest line in that project that had no task associated.
- All kanban, kanban mobile, list and form views are covered.
- I removed, BTW, some Bootstrap 3 and bad replacements in some views.

---

@Tecnativa TT19205
Inspired by https://github.com/OCA/timesheet/pull/139.

### CODE REVIEW

File: project_timesheet_time_control/models/account_analytic_line.py
Comment (Tardo) [inline project_timesheet_time_control/models/account_analytic_line.py]:
```suggestion
        return super().create(self._eval_date(vals))
```

File: project_timesheet_time_control/models/account_analytic_line.py
Comment (Tardo) [inline project_timesheet_time_control/models/account_analytic_line.py]:
```suggestion
        return super().write(self._eval_date(vals))
```

**Review by chienandalu — COMMENTED** [review]
These improvements are great!

A little issue:

- Go to timesheet and create two lines succesively.
- The timer of the first one doesn't stop so the times overlap.

Is it intended?

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
> A little issue:
> 
>     * Go to timesheet and create two lines succesively.
> 
>     * The timer of the first one doesn't stop so the times overlap.
> 
> 
> Is it intended?

There's no constraint that forbids creating 2 lines with 0 time. Maybe I should add a unique constraint to avoid that?

**Comment by pedrobaeza** [comment]
I don't think we should control that.

**Comment by yajo** [comment]
That's what I though at first also. After having these controls in place, it's more likely that you create your lines from projects/tasks view than timesheet ones.

:thinking: Maybe I could add a button to create a line from scratch from the menu, opening the wizard without smar fills, to let user autostop latest line.

**Comment by pedrobaeza** [comment]
Uhm, maybe we can make a similar screen to the attendance one with big button for starting and stopping.

**Comment by yajo** [comment]
That'd be much harder than just adding a menu item that opens the wizard. It would involve JS. I think the menu item should be enough

**Comment by pedrobaeza** [comment]
Well, let's put it on the roadmap. I don't see the menu too much added value.

**Comment by yajo** [comment]
I thought about it and it makes sense. I added this menu: *Timesheets > Timesheet > Start work*

It opens this wizard, without autocompletes. In the screenshot you can see the obvious benefit of doing it this way instead of just creating a new record: it warns about an old running timer and lets you go to it or stop it and start a new timer with one click:

Captura de pantalla de 2019-10-01 12-31-38


So, although I think we will use it less because the new buttons in tasks and projects are so cool, I still think it's good to have it for those that want to input a new line from scratch.

After all, it's just an action and menu item, so easy it's done already.

**Review by Tardo — APPROVED** [review]
LGTM :+1: Awesome improvements!

**Comment by pedrobaeza** [comment]
Please add Spanish translation

**Comment by yajo** [comment]
Done. I just have to add some tests and try to ensure the wizard always features a hotkey to make it more handy.

**Comment by yajo** [comment]
> try to ensure the wizard always features a hotkey

Enforcing a hotkey is very error-prone in a wizard, it's easier to rely on the automatic one, as far as there are available, so I opened https://github.com/OCA/timesheet/pull/278 and patched a little bit kanban and tree views here to add more chances for the wizard button to get a hotkey. Now it will have one almost always.

**Comment by pedrobaeza** [comment]
Can you improve a bit more test coverage?

**Comment by yajo** [comment]
Yes, it's on my roadmap, do not merge still please.

**Review by rafaelbn — APPROVED** [review]
Awesome! Functionally :+1:

**Comment by yajo** [comment]
If I'm not wrong, this should go 100% :umbrella: now.

**Comment by yajo** [comment]
Please do not merge yet. I still have to change one thing to make it compatible with the upcoming patch for `crm_timesheet`.

**Comment by yajo** [comment]
Ready to merge.

**Comment by rafaelbn** [comment]
@Yajo travis is :red_circle:

**Comment by yajo** [comment]
:heavy_check_mark: now.

**Comment by yajo** [comment]
Once merged let's continue with https://github.com/OCA/timesheet/pull/279.

**Pull request merged**
